src/expression/definitions/interpolate.ts

```diff
@@ -170,6 +173,10 @@ class Interpolate implements Expression {
 
         const outputLower = outputs[index].evaluate(ctx);
         const outputUpper = outputs[index + 1].evaluate(ctx);
+        
+        if (this.type.kind == 'projection') {
```

Comment:
I think this can be added to the below switch-case?

Reply:
A type guard is needed, so it turns the switch into:

```ts
        switch (this.operator) {
            case 'interpolate':
                if (this.type.kind !== 'projection') {
                    return interpolate[this.type.kind](outputLower, outputUpper, t);
                }
            case 'interpolate-hcl':
                return interpolate.color(outputLower, outputUpper, t, 'hcl');
            case 'interpolate-lab':
                return interpolate.color(outputLower, outputUpper, t, 'lab');
            case 'interpolate-projection': {
                return interpolate.projection(outputLower, outputUpper, t);
            }
        }
    ```

Reply:
This is why.

<img width="976" alt="Screenshot 2024-11-13 at 08 44 34" src="https://github.com/user-attachments/assets/a1fe7582-4ec3-44a0-8c9c-5924d9d8b514">

Reply:
The stupid part is that this, or insertion of any of the other possible string values for kind, doesn't have a type problem

<img width="611" alt="Screenshot 2024-11-13 at 08 51 59" src="https://github.com/user-attachments/assets/513028a0-6cb3-4634-acd2-5037f207bb6a">

Reply:
```ts
case 'interpolate':
      return interpolate[this.type.kind.toString()](outputLower, outputUpper, t);
  case 'interpolate-hcl':
      return interpolate.color(outputLower, outputUpper, t, 'hcl');
  ```
  
  A `kind.toString()` resolves it

Reply:
I think this removes the type guard from the `interpolate` object, which I'm not sure is what we want.
I generally try to avoid using `object[property]` as much as possible due to these limitation and the fact that it's harder to read the code this way.

Reply:
I like this one better to be honest.

Reply:
By the way, if this is possible for other types, why do we need an extra case for interpolate-projection then?